pkg/quota/v1/evaluator/core/resource_claims.go

```diff
@@ -0,0 +1,165 @@
+func (p *claimEvaluator) Constraints(required []corev1.ResourceName, item runtime.Object) error {
+	// no-op for resource claims
+	return nil
+}
+
+// GroupResource that this evaluator tracks
+func (p *claimEvaluator) GroupResource() schema.GroupResource {
+	return resourceapi.SchemeGroupVersion.WithResource("resourceclaims").GroupResource()
+}
+
+// Handles returns true if the evaluator should handle the specified operation.
+func (p *claimEvaluator) Handles(a admission.Attributes) bool {
+	op := a.GetOperation()
+	if op == admission.Create {
+		return true
+	}
+	if op == admission.Update {
+		return true
+	}
+	return false
+}
+
+// Matches returns true if the evaluator matches the specified quota with the provided input item
+func (p *claimEvaluator) Matches(resourceQuota *corev1.ResourceQuota, item runtime.Object) (bool, error) {
+	return generic.Matches(resourceQuota, item, p.MatchingResources, generic.MatchesNoScopeFunc)
+}
+
+// MatchingScopes takes the input specified list of scopes and input object. Returns the set of scopes resource matches.
+func (p *claimEvaluator) MatchingScopes(item runtime.Object, scopes []corev1.ScopedResourceSelectorRequirement) ([]corev1.ScopedResourceSelectorRequirement, error) {
+	return []corev1.ScopedResourceSelectorRequirement{}, nil
+}
+
+// UncoveredQuotaScopes takes the input matched scopes which are limited by configuration and the matched quota scopes.
+// It returns the scopes which are in limited scopes but don't have a corresponding covering quota scope
+func (p *claimEvaluator) UncoveredQuotaScopes(limitedScopes []corev1.ScopedResourceSelectorRequirement, matchedQuotaScopes []corev1.ScopedResourceSelectorRequirement) ([]corev1.ScopedResourceSelectorRequirement, error) {
+	return []corev1.ScopedResourceSelectorRequirement{}, nil
+}
+
+// MatchingResources takes the input specified list of resources and returns the set of resources it matches.
+func (p *claimEvaluator) MatchingResources(items []corev1.ResourceName) []corev1.ResourceName {
+	result := []corev1.ResourceName{}
+	for _, item := range items {
+		if item == claimObjectCountName /* object count quota fields */ ||
+			item == corev1.ResourceClaims /* claim resources alias */ ||
+			strings.HasSuffix(string(item), corev1.ResourceClaimsPerClass /* by resource class */) {
+			result = append(result, item)
+		}
+	}
+	return result
+}
+
+// Usage knows how to measure usage associated with item.
+func (p *claimEvaluator) Usage(item runtime.Object) (corev1.ResourceList, error) {
+	result := corev1.ResourceList{}
+	claim, err := toExternalResourceClaimOrError(item)
+	if err != nil {
+		return result, err
+	}
+
+	// charge for claim
+	result[corev1.ResourceClaims] = *(resource.NewQuantity(1, resource.DecimalSI))
+	result[claimObjectCountName] = *(resource.NewQuantity(1, resource.DecimalSI))
+	for _, request := range claim.Spec.Devices.Requests {
+		if request.AdminAccess {
+			// No need to count, admins get unlimited access.
```

Comment:
Probably that is what will happen. But I'm not sure whether it's always going to be that way, so it seemed better to add this special case.

Reply:
Permission to use "admin access" might not be tied to the namespace. It's checked via VAP, so there will be plenty of opportunities to configure it differently, potentially so that admins run pods with admin access to devices in a namespace where normal users are limited by quota.